### DISCUSSION TITLE
How to disable nvim-cmp for command mode wildmenu?

### DISCUSSION THREAD

**Discussion by mildred** [hrsh7th/nvim-cmp#1731] [Q&A]
Hi,

I'm having a hard time understanding how to configure the command-line completion and I wish to disable nvim-cmp for the command mode (when completing commands like `:e`).

When running `nvim --clean` here is what I can do:

- `:e /usr/` `<Tab><Tab><Tab><Tab><Tab><Tab><Tab><Tab>` it completes to `/usr/local`
- `<Down>` it enters the `/usr/local` directory and suggest completions for `/usr/local/bin/`
- `<Up>` it suggest completions for `/usr/local/` back again

When running with nvim-cmp with no explicit mapping on arrow keys but a mapping on `Tab` as follows

```lua
    ['<Tab>'] = function(fallback)
      if cmp.visible() then
        cmp.select_next_item()
        cmp.mapping.complete()
      else
        fallback()
      end
    end,
```

When trying to achieve the same, I cannot:

- `:e /usr/` `<Tab><Tab><Tab><Tab><Tab><Tab><Tab><Tab><Tab>` it completes to `/usr/local` **ok**
- I see no way to enter the /usr/local directory and complete subdirectories. Arrow keys do not work
- no way to go up a directory either

Is there a way I could configure the nvim-cmp mapping, for that, or at least disable nvim-cmp for the command wildmenu?

Thank you

**Comment by mildred** [comment]
Well, I had this in my config that I did not see right away:

```lua
cmp.setup.cmdline(
    ":",
    {
        sources = cmp.config.sources(
            {
                {name = "path"}
            },
            {
                {name = "cmdline"}
            }
        ),
        mapping = cmp.mapping.preset.cmdline(),
    }
)
```

I removed this snippet and I got the raw cmdline again.

Still, it would be nice to be able to specify a mapping allowing to doe down and up in directories with nvim-cmp.

**Reply by mildred** [reply]
Instead I'm using https://vi.stackexchange.com/questions/22627/switching-arrow-key-mappings-for-wildmenu-tab-completion

**Comment by mildred** [comment]
Well, it does not work very well. After some time, I'm left without ANY completion in the command mode. Tab keystrokes just show `^I` characters around

edit: `:cmap` shows that the keybindings has been hijacked by `nvim-cmp`

**Comment by TM10YMhp** [comment]
This is what worked best for me:

```lua
vim.api.nvim_create_autocmd({ "CmdlineEnter" }, {
  pattern = { ":" },
  callback = function()
    local mappings = vim.api.nvim_get_keymap("c")
    for _, v in pairs(mappings) do
      if v.desc == "cmp.utils.keymap.set_map" then
        vim.keymap.del("c", v.lhs)
      end
    end
  end,
})
```
